lib/ansible/modules/cloud/cloudstack/cs_facts.py

```diff
@@ -121,7 +121,12 @@
 class CloudStackFacts(object):
 
     def __init__(self):
-        self.facts = ansible_facts(module)
+        collector = ansible_collector.get_ansible_collector(all_collector_classes=default_collectors.collectors,
+                                                            filter_spec='default_ipv4',
+                                                            gather_subset=['network'],
```

Comment:
once https://github.com/ansible/ansible/pull/27085 is merged, this could become
``` python
collector = ansible_collector.get_ansible_collector(all_collector_classes=default_collectors.collectors,
                                                            filter_spec='default_ipv4'
                                                            gather_subset=['!all', '!min', 'network'],
                                                            gather_timeout=10)
```

The '!all !min' will then prevent gathering the other 'minimal' facts and just grab the network facts.